TheJumpCloud/jumpcloud-ADMU
# Cut 3870 error mapping

## Issues
* [CUT-3870-Error-Mapping](https://jumpcloud.atlassian.net/browse/cut-3870) -CUT-3870-Error-Mapping

## What does this solve?
This change gives visibility to admins when an error occurs, they are provided with basic steps to troubleshoot the error that is encountered and/or to the troubleshooting errors wiki page.
## Is there anything particularly tricky?
n/a
## How should this be tested?
Backup-Error Scenario
1. Load NTUser.dat to registry
2. Run ADMU
3. This should trigger a backup error
![image](https://github.com/TheJumpCloud/jumpcloud-ADMU/assets/97972790/cc86305c-a401-41d4-aed4-62f519984a21)

Rename error scenario
1. Delete NTUser.dat once a .bak is created during the migration process
2. Validate that rename error is thrown
![image](https://github.com/TheJumpCloud/jumpcloud-ADMU/assets/97972790/952e0cd3-30bd-4bc7-af1e-ce0f72ccad05)

Load/Unload error scenario
1. Open Registry Editor
2. During migration click on the loaded hive in HKEY/Users
3. Validate that a load/unload error is thrown
![image](https://github.com/TheJumpCloud/jumpcloud-ADMU/assets/97972790/4c94a665-0419-428c-bcca-a2efb9fff07b)
